planning/mission_planner/src/mission_planner/mission_planner.hpp

```diff
@@ -68,86 +70,63 @@ class MissionPlanner : public rclcpp::Node
+    const Header & header, const std::vector<LaneletSegment> & segments, const Pose & goal_pose,
+    const UUID & uuid, const bool allow_goal_modification);
   LaneletRoute create_route(
-    const std_msgs::msg::Header & header, const std::vector<geometry_msgs::msg::Pose> & waypoints,
-    const geometry_msgs::msg::Pose & goal_pose, const bool allow_goal_modification);
-
-  RouteState::Message state_;
-  component_interface_utils::Publisher<RouteState>::SharedPtr pub_state_;
-  component_interface_utils::Publisher<Route>::SharedPtr pub_route_;
-  component_interface_utils::Publisher<NormalRoute>::SharedPtr pub_normal_route_;
-  component_interface_utils::Publisher<MrmRoute>::SharedPtr pub_mrm_route_;
-  void change_state(RouteState::Message::_state_type state);
-
-  component_interface_utils::Service<ClearRoute>::SharedPtr srv_clear_route_;
-  component_interface_utils::Service<SetRoute>::SharedPtr srv_set_route_;
-  component_interface_utils::Service<SetRoutePoints>::SharedPtr srv_set_route_points_;
-  component_interface_utils::Service<ChangeRoute>::SharedPtr srv_change_route_;
-  component_interface_utils::Service<ChangeRoutePoints>::SharedPtr srv_change_route_points_;
-  void on_clear_route(
-    const ClearRoute::Service::Request::SharedPtr req,
-    const ClearRoute::Service::Response::SharedPtr res);
-  void on_set_route(
-    const SetRoute::Service::Request::SharedPtr req,
-    const SetRoute::Service::Response::SharedPtr res);
-  void on_set_route_points(
-    const SetRoutePoints::Service::Request::SharedPtr req,
-    const SetRoutePoints::Service::Response::SharedPtr res);
-
-  component_interface_utils::Service<SetMrmRoute>::SharedPtr srv_set_mrm_route_;
-  component_interface_utils::Service<ClearMrmRoute>::SharedPtr srv_clear_mrm_route_;
-  void on_set_mrm_route(
-    const SetMrmRoute::Service::Request::SharedPtr req,
-    const SetMrmRoute::Service::Response::SharedPtr res);
-  void on_clear_mrm_route(
-    const ClearMrmRoute::Service::Request::SharedPtr req,
-    const ClearMrmRoute::Service::Response::SharedPtr res);
-
-  void on_change_route(
-    const SetRoute::Service::Request::SharedPtr req,
-    const SetRoute::Service::Response::SharedPtr res);
-  void on_change_route_points(
-    const SetRoutePoints::Service::Request::SharedPtr req,
-    const SetRoutePoints::Service::Response::SharedPtr res);
-
-  double reroute_time_threshold_{10.0};
-  double minimum_reroute_length_{30.0};
-  bool check_reroute_safety(const LaneletRoute & original_route, const LaneletRoute & target_route);
+    const Header & header, const std::vector<Pose> & waypoints, const Pose & goal_pose,
+    const UUID & uuid, const bool allow_goal_modification);
 
-  std::shared_ptr<LaneletRoute> normal_route_{nullptr};
-  std::shared_ptr<LaneletRoute> mrm_route_{nullptr};
+  rclcpp::TimerBase::SharedPtr data_check_timer_;
+  void check_initialization();
+
+  double reroute_time_threshold_;
+  double minimum_reroute_length_;
```

Comment:
better to set default value (same for other parts.
because the default value for double is not always 0.0

Reply:
```suggestion
  double reroute_time_threshold_{10.0};
  double minimum_reroute_length_{30.0};
```

these are removed default values

Reply:
These values are not used because the parameters are always assigned in the constructor.

Reply:
Yes! It looks OK this time and no need to modify this

We had recently made sure to initialize as well, because death could happen if you forget to pass in parameters.
FYI: https://star4.slack.com/archives/C03QW0GU6P7/p1694699548349669?thread_ts=1694534749.153109&cid=C03QW0GU6P7 (It is bad that the parameter is not set.)

Reply:
The problem is that the parameter is simply not assigned, and setting the initial value here is not the correct solution. I think it is better to avoid introducing unnecessary measures because bugs may be harder to find.

It's a good thing that nodes die and anomalies are easy to detect. I think it's more risky to keep working with unintended parameters.